src/applications/disability-benefits/all-claims/content/toxicExposure.jsx

```diff
@@ -42,3 +56,100 @@ export const conditionsDescription = (
 export const gulfWar1990PageTitle = 'Service locations after August 2, 1990';
 export const gulfWar1990Question =
   'Did you serve in any of these Gulf War locations on or after August 2, 1990? Check any locations where you served.';
+
+/**
+ * Builds the Schema based on user entered condition names
+ * 
+ * Example output:
+{
+    type: 'object',
+    properties: {
+      anemia: {
+        type: 'boolean'
+      },
+      tinnitusringingorhissinginears: {
+        type: 'boolean'
+      },
+      none: {
+        type: 'boolean'
+      }
+    }
+  }
+}
+ *
+ * @param {object} formData - Full formData for the form
+ * @returns {object} Object with id's for each condition
+ */
+export const makeTEConditionsSchema = formData => {
+  const options = (formData?.newDisabilities || []).map(disability =>
```

Comment:
What happens in the case of an empty array here? Would `disability.condition` need the safe navigation operator too? Curious if this be equivalent? `formData?.newDisabilities?.map(disability => `

 On a related note, I see some tests around null condition list for the toxicExposure spec, but I'm wondering if it'd be worthwhile to add a null condition example to the toxicExposureConditions spec?

Reply:
Good questions.

In the case of an empty array, there would be no items to loop through, so line 85 would not execute. That example could be another way of preventing the loop however in some wild chance newDisabilities was a different data structure and so returned truthy, we could be incorrectly trying to use an array function on a different data structure.

The null condition (name) scenario is an odd one. So far I've only seen this happening if user picks 'new' for claim type, adds one or more new conditions and then removes them all. In this scenario there is a validation error on the `/new-disabilities/add` page that will not let the user move to the next page, so not sure how they could get to the `toxic-exposure-conditions` page.  
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/127446042/827ec62a-0954-492f-bcaf-ddb86b29b7c2)

The only way I was able to get to the TE conditions page so far without any new conditions is by using the save in progress dev tool and this is what displayed 
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/127446042/d6248d02-b619-4ca8-9d49-e61d653b1ead)

I think if we wanted to prevent this scenario, we would need to update the `depends` clause on the form config to prevent access to this page entirely, but again, not clear how this could happen. Feels like a separate ticket?